plugins/jobtype/src/azkaban/jobtype/HadoopTuningSecurePigWrapper.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import static org.apache.hadoop.security.UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.security.PrivilegedExceptionAction;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.hadoop.mapred.Counters;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.log4j.Logger;
+import org.apache.pig.PigRunner;
+import org.apache.pig.tools.pigstats.JobStats;
+import org.apache.pig.tools.pigstats.PigStats;
+import org.apache.pig.tools.pigstats.PigStats.JobGraph;
+
+import azkaban.jobtype.tuning.TuningCommonConstants;
+import azkaban.jobtype.tuning.TuningParameterUtils;
+import azkaban.jobtype.tuning.TuningErrorHandler;
+import azkaban.utils.Props;
+
+/**
+ * This class represent wrapper for running pig job with tuning enabled.
+ */
+public class HadoopTuningSecurePigWrapper {
```

Comment:
This class needs to leverage `HadoopSecurePigWrapper`. There is a huge amount of code duplication here. Through refactoring if necessary, this class should only be adding logic for the tuning injections, not anything else.